integrationtests/base.go

```diff
@@ -0,0 +1,139 @@
+package integrationtests
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+)
+
+// UpWg used for setting up the wireguard interface
+func UpWg(wgConfig, wgConfDir string, t *testing.T) (string, error) {
+	tempdir := t.TempDir()
+	f, err := os.Create(tempdir + "/test.conf")
+	if err != nil {
+		return "", errors.Wrapf(err, "error creating file")
+	}
+	_, err = f.WriteString(wgConfig)
+	if err != nil {
+		return "", errors.Wrapf(err, "error writing file")
+	}
+	cmd := exec.Command("wg-quick", "up", f.Name())
+	_, err = cmd.Output()
+	if err != nil {
+		return "", errors.Wrapf(err, "error executing wg-quick up")
+	}
+
+	return f.Name(), nil
+
+}
+
+// DownWG used for tearing down the wireguard interface
+func DownWG(conf string) (string, error) { //tempdir
+	cmd := exec.Command("wg-quick", "down", conf)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrapf(err, "error executing wg-quick down")
+	}
+	return string(out), nil
+}
+
+// RemoteRun used for running cmd remotly using ssh
+func RemoteRun(user string, addr string, cmd string, privateKey string) (string, error) {
+	key, err := ssh.ParsePrivateKey([]byte(privateKey))
+	if err != nil {
+		return "", errors.Wrapf(err, "error parsing ssh private key %v", key)
+	}
+	// Authentication
+	config := &ssh.ClientConfig{
+		User:            user,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(key),
+		},
+	}
+
+	// Connect
+	port := "22"
+	client, err := ssh.Dial("tcp", net.JoinHostPort(addr, port), config)
+	if err != nil {
+		return "", errors.Wrapf(err, "error starting ssh connection ")
+	}
+	session, err := client.NewSession()
+	if err != nil {
+		return "", errors.Wrapf(err, "error creating new session")
+	}
+	defer session.Close()
+	var b bytes.Buffer
+	session.Stdout = &b
+	// trying to make compound output work insted of buffer
+
+	err = session.Run(cmd)
+	if err != nil {
+		return "", errors.Wrapf(err, "error executing command on remote")
+	}
+	return b.String(), nil
+}
+
+// TODO: investigate if this is needed
+func IsIPReachable(wgConfig string, IPCheckList []string, privateKey string) error {
```

Comment:
1- parameters should be camel cased so ipsToCheck instead of IPCheckList
2- the function talks about a singular IP `IsIPReachable` so please make it `IsIPReachable(wgConfig , ip, privateKey string)` all of type string, no need to work against a string slice
3- if you have multiple IPs you want to check against, you can do such in goroutines

Reply:
I checked we are using this function only in the wireguard test i can replace it with the regular goroutines and use loops to check on the machines ip

Reply:
can we use list of ips in one call instead of ip and have the go routines inside the function?